flask_monitoringdashboard/core/alert/chat.py

```diff
@@ -0,0 +1,112 @@
+import requests
+
+from flask_monitoringdashboard.core.alert.alert_content import AlertContent
+
+SLACK_CHAR_LIMIT = 2750
+ROCKET_CHAT_CHAR_LIMIT = 4500
+TEAMS_CHAR_LIMIT = 5000
+
+
+def send_message(alert_content: AlertContent):
+    from flask_monitoringdashboard import config
+
+    payload_creators = {
+        "SLACK": create_slack_payload,
+        "ROCKET_CHAT": create_rocket_chat_payload,
+        "TEAMS": create_teams_payload,
+    }
+
+    creator = payload_creators.get(config.chat_platform)
+    if not creator:
+        print("Invalid chat platform.")
+        return
+
+    payload = creator(alert_content)
+
+    try:
+        resp = requests.post(config.chat_webhook_url, json=payload, timeout=5)
+        resp.raise_for_status()
+    except Exception as e:
+        print("Alert delivery failed:", e)
+
+
+def create_slack_payload(alert_content: AlertContent):
+    return {
+        "blocks": [
+            {
+                "type": "section",
+                "text": {"type": "mrkdwn", "text": alert_content.create_body_mrkdwn(SLACK_CHAR_LIMIT)}
+            }
+        ]
+    }
+
+
+def create_rocket_chat_payload(alert_content: AlertContent):
+    return {
+        "text": alert_content.create_body_markdown(ROCKET_CHAT_CHAR_LIMIT),
+    }
+
+
+def create_teams_payload(alert_content: AlertContent):
+    return {
+        "type": "message",
```

Comment:
haha. Why am I not surprised that the Microsoft alternative is the most complicated one?